### PR TITLE
Created a route for Purchased items deails

### PR DESCRIPTION
**Pull request by 201560543**

### CODE REVIEW

File: app/web/api/purchase_routes.py [hunk 1]
```diff
@@ -105,5 +104,27 @@ def get_all_purchases():
                     'timeline': timeline_groups
                 }
 
-
     return json.dumps(result_dict, default=converter)
+
+
+@purchase.route('/purchase_item_details', methods=['GET'])
+@exception_handler(custom_msg="Issues in fetching purchase_item_details")
+def get_all_markets():
```
Comment (PansulBhatt) [inline app/web/api/purchase_routes.py:112]: Can you change this function name?

File: app/web/api/account_routes.py [hunk 2]
```diff
@@ -66,6 +71,28 @@ def get_account_details(return_json=True):
     return result_dict
 
 
+@account.route("/delete-user", methods=["DELETE"])
+@exception_handler()
+@requires_auth
+@check_parameters(required_params=['person_id'], method_type='DELETE')
+def delete_user():
```
Comment (PansulBhatt) [inline app/web/api/account_routes.py:78]: This should be in person-routes? Right @pdeguzman96

File: app/web/api/account_routes.py [hunk 3]
```diff
@@ -19,28 +19,33 @@
 from datetime import datetime as dt
 from web.constants import ADD_ACCOUNT_PARAMS, NEW_ACCOUNT_PARAMS
 
+
 @account.route('/user-details', methods=['GET'])
 @exception_handler(custom_msg='Issues in fetching')
 @requires_auth
-@check_parameters(required_params=['email','auth0_id'])
+@check_parameters(required_params=['email', 'auth0_id'])
```
Comment (PansulBhatt) [inline app/web/api/account_routes.py:26]: Should we push this to person routes now?

File: app/web/api/account_routes.py (hunk 2 above)
Comment (201560543) [reply app/web/api/account_routes.py:78]: Should I move both user-details and delete-user to the Person route?

Comment (PansulBhatt) [reply app/web/api/account_routes.py:78]: Yes

File: app/web/api/invoice_routes.py [hunk 4]
```diff
@@ -26,8 +27,13 @@ def get_all_invoice_headers(return_json=True):
...
             # checking if the mandatory parameter has been passed.
-            if account_number is None:
-                results = memsql_session.query(InvHeaderTbl.table).all()
+            if account_number and organization_id is None:
+                # If we don't have any of the params then we should send a 400 error
+                abort(400, description="Account number or Organization ID are required.")
+
+            # Here if account_number is passed we filter on account number but if it's not present we expect the organization ID to be present.
+            if not account_number:
+                results = memsql_session.query(InvHeaderTbl.table).filter_by(organization_id=organization_id)
```
Comment (pdeguzman96) [inline app/web/api/invoice_routes.py:36]: Unfortunately this won't work either. So we have 2 options for when a user selects "All Locations":

1. User gives `organization_id`. Using this, we get all related organizations using the `organization` table, then get all their accounts from the `account` table. Then we get all invoices that have `invoice.account_number` equal to one of those accounts.
2. User selects "All Locations" and Subodh just gives us an array of account numbers, then we get all invoices that have `invoice.account_number` equal to one of those accounts.

I'm sorry for how unclear this was and changing the approach, but let's actually just go with option 2. Subodh can just send us an array of account numbers when a user clicks "All Locations".

File: app/web/api/person_routes.py [hunk 5]
```diff
@@ -99,6 +100,45 @@ def delete_user():
     return json.dumps(body, default=converter), 200
 
 
+@person.route('/deleted-users', methods=['GET'])
+@exception_handler(custom_msg="Issues in fetching all deleted persons")
+def get_all_deleted_people(return_json=True):
+    """
+    API to ge a list of al the people who had been soft deleted
+    """
+    results = db.session.query(Person).all()
```
Comment (pdeguzman96) [inline app/web/api/person_routes.py:109]: Should we return all from Person table or filter by `organization_id` so Subodh only gets users in the organization?

File: app/web/api/person_routes.py [hunk 6]
```diff
@@ -99,6 +100,45 @@ def delete_user():
...
+def reactivate_user():
+    """
+    Re activates the person instance by person ID.
+    returns: 200 with success status
+    """
+    body = {
+        'success': None
+    }
+    # Get the person instance by making a call to Person instance.
+    person_id = request.form.get("person_id")
```
Comment (pdeguzman96) [inline app/web/api/person_routes.py:132]: Update this to a request.json.get

Comment (201560543) [reply app/web/api/person_routes.py:132]: any specific reason for json.get?
I mean both form.get and json.get do the same things and in a POST request it's better to use form.get by convention.

Comment (pdeguzman96) [reply app/web/api/person_routes.py:132]: Ah it was just for consistency. All our other POST and PUTs use json.get

Comment (pdeguzman96) [reply app/web/api/person_routes.py:132]: Also I was on a call with Subodh yesterday and he didn't know how to send DELETE requests using form data hahaha. He tried many ways of formatting his API calls, and I just gave in and updated the DELETE to use json instead. Can we stick with request.json for POSTs, PUTs, and DELETEs?

**Pull request merged**